npm_nodejs/cdns.md

```diff
@@ -49,7 +49,7 @@ accessing npm packages as ES Module bundles. esm.sh uses
 ensure that it is consumable as an ES Module. In many cases you can just import
 the npm package into your Deno application:
 
-```ts
+```tsx ignore
```

Comment:
Not happy that I had to ignore all examples in this file - this is probably an error and we should try to fix that. @kitsonk would you mind taking a look?

`deno test --doc --unstable --import-map=.github/import_map.json npm_nodejs/`
produces following errors if these codeblocks are not ignored:
```
error: TS2300 [ERROR]: Duplicate identifier 'LibraryManagedAttributes'.
        type LibraryManagedAttributes<C, P> = C extends React.MemoExoticComponent<infer T> | React.LazyExoticComponent<infer T>
             ~~~~~~~~~~~~~~~~~~~~~~~~
    at https://cdn.esm.sh/v45/@types/react@17.0.16/index.d.ts:3058:14

    'LibraryManagedAttributes' was also declared here.
            type LibraryManagedAttributes<C, P> = C extends React.MemoExoticComponent<infer T> | React.LazyExoticComponent<infer T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~
        at https://cdn.esm.sh/v45/@types/react@17.0.17/index.d.ts:3060:14

TS2300 [ERROR]: Duplicate identifier 'LibraryManagedAttributes'.
        type LibraryManagedAttributes<C, P> = C extends React.MemoExoticComponent<infer T> | React.LazyExoticComponent<infer T>
             ~~~~~~~~~~~~~~~~~~~~~~~~
    at https://cdn.esm.sh/v45/@types/react@17.0.17/index.d.ts:3060:14

    'LibraryManagedAttributes' was also declared here.
            type LibraryManagedAttributes<C, P> = C extends React.MemoExoticComponent<infer T> | React.LazyExoticComponent<infer T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~
        at https://cdn.esm.sh/v45/@types/react@17.0.16/index.d.ts:3058:14

TS2717 [ERROR]: Subsequent property declarations must have the same type.  Property 'source' must be of type 'DetailedHTMLProps<SourceHTMLAttributes<HTMLSourceElement>, HTMLSourceElement>', but here has type 'DetailedHTMLProps<SourceHTMLAttributes<HTMLSourceElement>, HTMLSourceElement>'.
            source: React.DetailedHTMLProps<React.SourceHTMLAttributes<HTMLSourceElement>, HTMLSourceElement>;
            ~~~~~~
    at https://cdn.esm.sh/v45/@types/react@17.0.17/index.d.ts:3163:13

    'source' was also declared here.
                source: React.DetailedHTMLProps<React.SourceHTMLAttributes<HTMLSourceElement>, HTMLSourceElement>;
                ~~~~~~
        at https://cdn.esm.sh/v45/@types/react@17.0.16/index.d.ts:3161:13

```

Reply:
How does it work, is it loading all the doc blocks in the same context?  The errors you are posting looks like two different version of React typings are being included by esm.sh, which is not an issue with the code, more of an issue with how esm.sh is right now (and the risk of using an unpinned import, maybe we just pick a version?)

Reply:
> How does it work, is it loading all the doc blocks in the same context?

Same "program_state" yes, passed in as a collection of roots to `prepare_module_graph`.
Doing them one by one would probably mitigate this.

Reply:
Okay, so it's actually a problem with docstring typechecking that they end up as a "single program". @caspervonb could you look into fixing it?

Reply:
Yup!